blog/2022-04-25-jwt-vs-api-key-authentication/index.md

```diff
@@ -0,0 +1,109 @@
+- JWT tokens should be kept relatively small in size (i.e., you don't want to put a list of all resources the client is authorized to access in the JTW token itself).
+- When JWT tokens expire, the consumer must request a new token using the Client ID and Secret value.
+- Identity Providers often charge based on the number of tokens issued.
+
+## API Key Authentication
+
+With API authentication, each client receives a unique secret key. Unlike JWT tokens, the key itself doesn't contain any actual data, it is simply a unique string associated with the client. Furthermore, there is no standard protocol for API Key authentication like OAuth, etc., so each implementation can differ.
+
+The simplest example of API token authentication is an environment variable that is checked against each API request.
+
+```js
+function myApiHandler(request) {
+  if (request.headers.get("API-Key") !== process.env.API_KEY) {
+    throw new Error("Not authorized");
+  }
+  // ...
+}
+```
+
+However, most API Key authentication systems offer more advanced management of keys. For example, an API Gateway could offer a self-serve portal where end-users issue tokens based on permissions and configuration that the API developer has set. Tokens can be issued with various permissions and with custom expirations times.
+
+A typical API Key authentication system will validate each key as it comes in with a request. If the key is valid, then data is returned with that key. This data could be identity information, permissions, and even details about their allowed use of the API, such as rate limits or IP restrictions.
+
+```
+function myApiHandler(request) {
+  const apiKey = request.headers.get("API-Key");
+  const apiKeyInfo = apiKeyService.validate(apiKey);
+
+  // Check various properties of the api key info
+  if (clientInfo?.accountId) {
+    // ...
+  }
+}
+```
+
+### Layered Permissions
+
+One of the most powerful capabilities of API Key authentication is the ability to build "layered" permission capabilities into your API. For example, you, as the API owner, could authorize a customer to create API Keys with permissions like `read:products` and `write:products`. The customer could go into their self-serve portal and issue a token with both permissions, but they could also issue a token with a sub-set of permissions. The benefit of this layered approach is that each token has only the permissions needed for its specific function.
+
+![Token Scopes](./token_scopes.gif)
+
+### Considerations of API Key Auth
+
+The main difference between API Key auth and JWT token auth is that the JWT Token is self-contained - the information asserted by the token is in the token. Whereas with an API Key the asserted information is stored in an external system. The externalization of assertion data makes API Keys more flexible for certain scenarios.
+
+- Simpler for end-users to use and understand.
```

Comment:
not super clear which method you're referring to here. I would state "pros/cons of API key auth:"
: